src/escape.c

```diff
@@ -92,7 +105,7 @@ void checkEscape(Scope *sc, Expression *e)
         {
             for (size_t i = 0; i < e->exps->dim; i++)
             {
-                checkEscape(sc, (*e->exps)[i]);
+                visit((*e->exps)[i]);
```

Comment:
This is most likely why it's failing - no subexpressions are actually getting checked.